js/messaging.js

```diff
@@ -233,12 +237,42 @@ function openURL({
       url.replace(/%2F.*/, '*').replace(/#.*/, '') :
       url.replace(/#.*/, '');
 
-  return queryTabs({url: urlQuery, currentWindow}).then(maybeSwitch);
+  return manageMatch || editMatch ? queryTabs().then(maybeSwitch) :
+  queryTabs({url: urlQuery, currentWindow}).then(maybeSwitch);
 
   function maybeSwitch(tabs = []) {
     const urlWithSlash = url + '/';
     const urlFF = FIREFOX && url.replace(/%2F/g, '/');
-    const tab = tabs.find(({url: u}) => u === url || u === urlFF || u === urlWithSlash);
+    const urlOptions = manageMatch ? URLS.ownOrigin + 'manage.html?#stylus-options' : null;
+    const urlManage = manageMatch ? URLS.ownOrigin + 'manage.html' : null;
+    let tab = tabs.find(({url: u}) => u === url || u === urlFF || u === urlWithSlash || u === urlOptions || u === urlManage);
+    if (!tab && prefs.get('openEditInWindow') && chrome.windows && editMatch) {
+      chrome.windows.create(
+        Object.assign({
+          url: url
+        }, prefs.get('windowPosition', {}))
+      );
+      return;
+    }
+    if (manageMatch) {
+      if (tab) {
+        const toggleOptions = url === urlOptions ? 'options-open' : 'options-close';
+        chrome.tabs.sendMessage(tab.id, {
+          'name': 'options',
+          'data': toggleOptions
+        });
+      }
+      getActiveTab()
+        .then(currentTab => {
+          const closePopup = tab && FIREFOX && currentTab.windowId !== tab.windowId ? false : true;
```

Comment:
eslint: Unnecessary use of boolean literals in conditional expression

may be replaced with `!(tab && FIREFOX && currentTab.windowId !== tab.windowId)` or `!tab || !FIREFOX || currentTab.windowId === tab.windowId`